src/core/logical_layers/renderers/BivariateRenderer.tsx

```diff
@@ -102,17 +125,151 @@ export class BivariateRenderer extends LogicalLayerDefaultRenderer {
     if (layerData == null) return;
     this.mountBivariateLayer(map, layerData, legend);
 
+    const clickHandler = (ev: maplibregl.MapMouseEvent & maplibregl.EventData) => {
+      const features = ev.target
+        .queryRenderedFeatures(ev.point)
+        .filter((f) => f.source.includes(this._sourceId));
+      if (!features.length || !legend || !features[0].geometry) return true;
+      const geometry = features[0].geometry;
+
+      const fillColor: FillColor = features[0].layer.paint?.['fill-color'];
+      if (!fillColor) return true;
+
+      const rgba = this.convertFillColorToRGBA(fillColor);
+      const cells: BivariateLegendStep[] = invertClusters(legend.steps, 'label');
+      const cellIndex = cells.findIndex((i) => i.color === rgba);
+
+      currentTooltipAtom.setCurrentTooltip.dispatch({
+        popup: (
+          <MapHexTooltip
+            cellLabel={cells[cellIndex].label}
+            cellIndex={cellIndex}
+            axis={legend.axis}
+            hexagonColor={rgba}
+          />
+        ),
+        popupClasses: { popupContent: popupContentRoot },
+        position: {
+          x: ev.originalEvent.clientX,
+          y: ev.originalEvent.clientY,
+        },
+        onClose(_e, close) {
+          close();
+          cleanSelection();
+        },
+        onOuterClick(_e, close) {
+          close();
+          cleanSelection();
+        },
+      });
+
+      if (!map.getSource('hex-selected-source')) {
+        map.addSource('hex-selected-source', {
+          type: 'geojson',
+          data: {
+            type: 'Feature',
+            geometry,
+            properties: {},
+          },
+        });
+
+        map.addLayer({
+          id: 'hex-selected-layer',
+          type: 'line',
+          source: 'hex-selected-source',
+          layout: {},
+          paint: {
+            'line-color': ACTIVE_HEXAGON_BORDER,
+            'line-width': 1,
+          },
+        });
+      } else {
+        const source = map.getSource('hex-selected-source') as mapboxgl.GeoJSONSource;
+        source.setData({
+          type: 'Feature',
+          geometry,
+          properties: {},
+        });
+      }
+
+      return true;
+    };
+
+    const mousemoveHandler = throttle(
+      (ev: maplibregl.MapMouseEvent & maplibregl.EventData) => {
+        const features = ev.target.queryRenderedFeatures(ev.point);
+        if (
+          !features.length ||
+          !features[0].geometry ||
+          !features[0]?.source.includes(this._sourceId) // we want to process hover only if top layer is bivariate
```

Comment:
can we get several features on hover? can feature[0] source be something else then `this._sourceId`, but feature[1] has `this._sourceId` as it's source?

Reply:
Yep, it's actually a fix for such situaltion. I can be with some layers from category: Layers in selected area. 
When select some of them, it becomes on top of bivariate layer and hover shouldn't work there.